latch/registry/utils.py

```diff
@@ -0,0 +1,355 @@
+import json
+import keyword
+import re
+from dataclasses import fields, is_dataclass
+from enum import Enum, EnumMeta
+from typing import (
+    Annotated,
+    Dict,
+    List,
+    Optional,
+    Type,
+    TypeVar,
+    Union,
+    get_args,
+    get_origin,
+)
+
+import latch.gql as gql
+from latch.types import LatchDir, LatchFile
+
+T = TypeVar("T")
+
+account_root_regex = re.compile("^account_root\/(?P<root>[^/]+)(?P<path>\/.*)$")
+mount_regex = re.compile("^mount\/(?P<mount>[^/]+)(?P<path>\/.*)$")
+
+shared_regex = re.compile("^shared(?P<path>\/.*)$")
+
+
+def clean(name: str):
+    # https://stackoverflow.com/a/3305731
+    cleaned = re.sub("\W|^(?=\d)", "_", name)
```

Comment:
pre-compile the regex

Reply:
does this work with unicode? `\W` is typically too restrictive
you also probably need `re.UNICODE` here?